### ISSUE TITLE
Debian Package

### ISSUE THREAD

**Issue by barak** [JaGoLi/ytdl-gui#32]
Just filed an ITP to push this into Debian.
Saw the Ubuntu PPA, took description etc from there, but redid debian/rules and such.
Prelim packaging on salsa.debian.org, and also in branch `debian` on fork `barak` of this repo.
Please let me know if that's okay with you, if you'd like to co-maintain, or any other concerns or suggestions.

Cheers,

--Barak Pearlmutter <barak@pearlmutter.net>

**Comment by JaGoLi** [comment]
This is great! It is already a bit of work for me to maintain this software project and packages for ubuntu and arch. Help with packaging is always appreciated. Can you send me some instructions to test out the debian repository?

**Comment by PranavBhattarai** [comment]
I think **flatpak** will solve the majority of issues like this for all distro if we package them.

**Comment by barak** [comment]
Okay, great.

Instructions? Sure. From the `debian` branch of https://github.com/barak/ytdl-gui
```
$ dpkg-checkbuilddeps
```
and install any missing dependencies, then
```
$ env DEB_RULES_REQUIRES_ROOT=no debian/rules binary
$ sudo dpkg --install ../youtubedl-gui_2.5-1_amd64.deb
```
I'll go ahead and upload it to Debian, since getting through the “new” queue takes forever. Once it's in the queue, you could test it by manually downloading and installing the binary package from https://ftp-master.debian.org/new/youtubedl-gui_2.5-1.html

**Comment by JaGoLi** [comment]
How are you going to handle the continually-updating dependency of youtube-dl? Before I was using cmake, I had a postinstall script that installed youtube-dl by pip3 so that users could get the latest version, otherwise the application wouldn't run. Also since youtube-dl is a moving target, I'm not sure if flatpak is the way to go. If I remember correctly it needs static dependencies.

**Comment by advaithm** [comment]
hey hoping on this thread may I build this for flatpaks, I've packaged a gitahead, this shouldn't be too hard too

**Comment by barak** [comment]
There are two questions there I guess.

1. Is the installed version of `youtube-dl` compatible with this version of `youtubedl-gui`?
2. Is the installed version of `youtube-dl` reasonably function with the web site it interacts with?

I was pretty much assuming that neither of those are `youtubedl-gui`'s problem. Debian right now has `youtube-dl` version `2021.02.10-1`, so I figured that would suffice. But of course it would be better to have a versioned dependency. And maybe runtime detection of available options. Would certainly welcome advice.

**Comment by advaithm** [comment]
@JaGoLi I've made a flatpak, may I submit it to flathub?

**Comment by advaithm** [comment]
> There are two questions there I guess.
> 
> 1. Is the installed version of `youtube-dl` compatible with this version of `youtubedl-gui`?
> 2. Is the installed version of `youtube-dl` reasonably function with the web site it interacts with?
> 
> I was pretty much assuming that neither of those are `youtubedl-gui`'s problem. Debian right now has `youtube-dl` version `2021.02.10-1`, so I figured that would suffice. But of course it would be better to have a versioned dependency. And maybe runtime detection of available options. Would certainly welcome advice.

It should be possible to add a check if  a uptodate version of youtube-dl is installed and prompt the user to update it via pip if required

**Comment by JaGoLi** [comment]
@advaithm Yeah sure, go ahead with the flathub package! 
@barak This application does not include `youtube-dl` it requires it as a dependency. The application is designed to work with any version of it, but older versions of `youtube-dl` are broken and can't interface with youtube APIs properly. This is why most users on debian or ubuntu need to install it via `pip`.

**Comment by advaithm** [comment]
we have a working build made by the build-bot https://github.com/flathub/flathub/pull/2312#issuecomment-832368647 we now have to wait for it to be merged in

**Comment by barak** [comment]
@JaGoLi Right, the package that I generated has an unversioned dependency on youtube-dl, so if a user tries to install the youtubedl-gui package it will automatically pull in youtube-dl. Sophisticated users could install youtube-dl in other ways (like a more recent version in /usr/local/bin/) and it should be okay. So it sounds like right now this is about as solved as possible, given the limitations of the Debian release cycle.
